### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Timi2424**
[DEMO LINK](https://Timi2424.github.io/react_todo-app-loading-todos/)

### CODE REVIEW

**Comment by aureliagorzna** [comment]
Looks good

**Review by aureliagorzna — APPROVED** [review]

File: src/components/TodoItem/TodoItem.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = React.memo((props) => {
```
Comment (mpiskadlo) [inline src/components/TodoItem/TodoItem.tsx:10]: I doubt you need `memo` here, same thing in `TodoList.tsx`

https://react.dev/reference/react/memo#should-you-add-memo-everywhere

File: src/components/Error/Error.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import cn from 'classnames';
+import { ErrorType } from '../../types/Errors';
+
+type Props = {
+  errorMessage: ErrorType | null,
+  setErrorMessage: (error: ErrorType | null)=> void;
```
Comment (mpiskadlo) [inline src/components/Error/Error.tsx:7]: You are using this `ErrorType | null` in multiple places. Create a `type` for it and import it wherever you need it

File: src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          data-cy="FilterLinkAll"
+          className={cn(
+            'filter__link',
+            { selected: filterBy === FilterBy.All },
+          )}
+          onClick={() => setFilterBy(FilterBy.All)}
```
Comment (mpiskadlo) [inline src/components/Footer/Footer.tsx:31]: I recommend creating such a handling function

```javascript
const hadnleFilterChange = (filer: Filterby) => () => {
  setFilterBy(filter);
  };
  
  // and then
  
  onClick={handleChangeFilter(FilterBy.All)}
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,10 +1,45 @@
...
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorType | null>(null);
+  const [filterBy, setFilterBy] = useState(FilterBy.All);
+
+  const visibleError = (error: ErrorType) => {
+    setErrorMessage(error);
+
+    setTimeout(() => setErrorMessage(null), 3000);
+  };
```
Comment (mpiskadlo) [inline src/App.tsx:24]: This function's name is a bit confusing

**Review by mpiskadlo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**